# International debt project

The goal of this project is to analyze the structure of the world external debt country-wise and the structure of each country external debt creditor-wise. Below you can find SQL queries generating reports that are used in Tableau dashbord.

First, I am looking through the database to get familiar with the dataset, then I check dataset cleanliness, and finally writing SQL queries.

World Bank Metadata Glossary on debt can be find here -> https://databank.worldbank.org/metadataglossary/international-debt-statistics/series/viewall

In [2]:
#Installing packages to work with Postgresql Server
!pip install ipython-sql
!pip install psycopg2
!pip install pgspecial
#loading sql magic
%load_ext sql

In [3]:
# Connecting to Postgresql server
%sql postgresql://jupyter:kjiflmyfgjvgt583!@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt

1. Let us look through the list of tables in the 'intdebt' DB and futher look through the 'international_debt' table's columns

In [4]:
%sql \d

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
2 rows affected.


[('public', 'international_debt', 'table', 'postgres'),
 ('public', 'international_debt_id_seq', 'sequence', 'postgres')]

In [5]:
%%sql
SELECT *
  FROM information_schema.columns
 WHERE table_schema = 'public'
   AND table_name   = 'international_debt';

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
6 rows affected.


[('intdebt', 'public', 'international_debt', 'id', 1, "nextval('international_debt_id_seq'::regclass)", 'NO', 'integer', None, None, 32, 2, 0, None, None, None, None, None, None, None, None, None, None, None, None, 'intdebt', 'pg_catalog', 'int4', None, None, None, None, '1', 'NO', 'NO', None, None, None, None, None, 'NO', 'NEVER', None, 'YES'),
 ('intdebt', 'public', 'international_debt', 'country_name', 2, None, 'YES', 'text', None, 1073741824, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 'intdebt', 'pg_catalog', 'text', None, None, None, None, '2', 'NO', 'NO', None, None, None, None, None, 'NO', 'NEVER', None, 'YES'),
 ('intdebt', 'public', 'international_debt', 'country_code', 3, None, 'YES', 'character varying', 3, 12, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 'intdebt', 'pg_catalog', 'varchar', None, None, None, None, '3', 'NO', 'NO', None, None, None, None, None, 'NO', 'NEVER', None, 'YES'),
 ('intdebt', 'public', 'international_debt', 'indicator_name', 4, None, 'YES', 'text', None, 1073741824, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 'intdebt', 'pg_catalog', 'text', None, None, None, None, '4', 'NO', 'NO', None, None, None, None, None, 'NO', 'NEVER', None, 'YES'),
 ('intdebt', 'public', 'international_debt', 'indicator_code', 5, None, 'YES', 'text', None, 1073741824, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 'intdebt', 'pg_catalog', 'text', None, None, None, None, '5', 'NO', 'NO', None, None, None, None, None, 'NO', 'NEVER', None, 'YES'),
 ('intdebt', 'public', 'international_debt', 'debt', 6, None, 'YES', 'numeric', None, None, None, 10, None, None, None, None, None, None, None, None, None, None, None, None, None, 'intdebt', 'pg_catalog', 'numeric', None, None, None, None, '6', 'NO', 'NO', None, None, None, None, None, 'NO', 'NEVER', None, 'YES')]

In [61]:
%%sql
SELECT * 
  FROM international_debt
 LIMIT 5;

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
5 rows affected.


[(1, 'Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 (2, 'Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 (3, 'Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 (4, 'Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 (5, 'Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001'))]

In [62]:
%%sql
SELECT * 
  FROM international_debt
 WHERE NOT (international_debt IS NOT NULL);

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
0 rows affected.


[]

There is one table with six columns and no NULL values in the 'intdebt' database.

Next steps are to understand the quantity of distinct countries and debt indicators, the most common debt indicators, and the relationships among different debt indicators within one country to correctly work with numerical data.

To understand the relationships among debt indicators, I will look through World Bank INTERNATIONAL DEBT STATISTICS 2021 report
https://openknowledge.worldbank.org/bitstream/handle/10986/34588/9781464816109.pdf

2. Finding the number of distinct countries

In [33]:
%%sql
SELECT COUNT(DISTINCT country_name) 
  FROM international_debt;

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
1 rows affected.


[(124,)]

3. Finding the number of distinct debt indicators and the most common debt indicators across countries

In [11]:
%%sql
SELECT indicator_code,
       indicator_name,
       COUNT(indicator_name) AS cnt_ind
  FROM international_debt
 GROUP BY indicator_code, indicator_name
 ORDER BY cnt_ind DESC;

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
25 rows affected.


[('DT.AMT.MLAT.CD', 'PPG, multilateral (AMT, current US$)', 124),
 ('DT.INT.MLAT.CD', 'PPG, multilateral (INT, current US$)', 124),
 ('DT.INT.OFFT.CD', 'PPG, official creditors (INT, current US$)', 124),
 ('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', 124),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', 124),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', 124),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', 123),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', 122),
 ('DT.INT.BLAT.CD', 'PPG, bilateral (INT, current US$)', 122),
 ('DT.AMT.BLAT.CD', 'PPG, bilateral (AMT, current US$)', 122),
 ('DT.DIS.MLAT.CD', 'PPG, multilateral (DIS, current US$)', 120),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', 113),
 ('DT.INT.PRVT.CD', 'PPG, private creditors (INT, current US$)', 98),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', 98),
 ('DT.INT.PCBK.CD', 'PPG, commercial banks (INT, current US$)', 84),
 ('DT.AMT.PCBK.CD', 'PPG, commercial banks (AMT, current US$)', 84),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', 79),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', 79),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', 69),
 ('DT.INT.PBND.CD', 'PPG, bonds (INT, current US$)', 69),
 ('DT.INT.PROP.CD', 'PPG, other private creditors (INT, current US$)', 54),
 ('DT.AMT.PROP.CD', 'PPG, other private creditors (AMT, current US$)', 54),
 ('DT.DIS.PRVT.CD', 'PPG, private creditors (DIS, current US$)', 53),
 ('DT.DIS.PCBK.CD', 'PPG, commercial banks (DIS, current US$)', 51),
 ('DT.DIS.PROP.CD', 'PPG, other private creditors (DIS, current US$)', 19)]

I found out that 'intdebt' DB covers 124 countries, debt structure varies across countries, and debt indicators have hierarchy according to the World Bank's report. However, 'intdebt' DB fails to show this hierarchy clearly because indicator_code does not show a full path of parent-child relationships.

Below you can find a picture that shows debt indicators hierarchy.

![Hierarchy image](https://raw.githubusercontent.com/Kirill4191/international-debt/main/hierarchy_diag.png)

There are three categories of debt indicators that represent the type of cash flow:
1. AMT - amortization (outflow)
2. INT - interest (outflow)
3. DIS - disbursement (inflow)

Each category of debt indicators has subcategories that represent the source of debt (creditors)

1. DPNG - Private nonguaranteed debt
2. OFFT - Official creditors
3. BLAT - Bilateral
4. MLAT - Multilateral
5. PRVT - Private
6. PBND - PPG debt: bonds
7. PCBK - PPG debt: commercial banks
8. PROP - Other private creditors

Three following debt indicators are root nodes that will show the structure of the world external debt country-wise:
1. DT.AMT.DLXF.CD - Principal repayments on external debt, long-term (AMT, current US$)

2. DT.INT.DLXF.CD - Interest payments on external debt, long-term (INT, current US$)

3. DT.DIS.DLXF.CD - Disbursements on external debt, long-term (DIS, current US$)

4. Finding the total debt inflow and outflow

In [13]:
%%sql
SELECT ROUND(SUM(debt), 2) AS total_debt_outflow_USD
  FROM international_debt
 WHERE indicator_code IN ('DT.AMT.DLXF.CD', 'DT.INT.DLXF.CD');

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
1 rows affected.


[(Decimal('936062666174.60'),)]

In [14]:
%%sql
SELECT ROUND(SUM(debt), 2) AS total_debt_inflow_usd
  FROM international_debt
 WHERE indicator_code = 'DT.DIS.DLXF.CD';

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
1 rows affected.


[(Decimal('264701069677.50'),)]

5. Ranking countries by the share in the world debt outflow and inflow

In [26]:
%%HTML
<div class='tableauPlaceholder' id='viz1623325047929' style='position: relative'><noscript><a href='#'><img alt='Dashboard 1 1000x800 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;In&#47;International_debt2&#47;Dashboard11000x800&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='International_debt2&#47;Dashboard11000x800' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;In&#47;International_debt2&#47;Dashboard11000x800&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1623325047929');                    var vizElement = divElement.getElementsByTagName('object')[0];                    if ( divElement.offsetWidth > 800 ) { vizElement.style.width='1000px';vizElement.style.height='827px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.width='1000px';vizElement.style.height='827px';} else { vizElement.style.width='100%';vizElement.style.height='1677px';}                     var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

In [22]:
%%sql
SELECT country_name,
       ROUND(SUM(debt), 2) AS debt_outflow,
       ROUND(SUM(SUM(debt)) OVER(), 2) AS world_debt_outflow,
       ROUND((SUM(debt) / (SUM(SUM(debt)) OVER()) * 100), 2) AS pct_world_debt_outflow,
       RANK() OVER(ORDER BY SUM(debt) DESC) AS rank_outflow
  FROM (
        SELECT *
          FROM international_debt
         WHERE indicator_code IN ('DT.AMT.DLXF.CD', 'DT.INT.DLXF.CD')
  ) AS subquery
 GROUP BY country_name
 ORDER BY debt_outflow DESC
 LIMIT 10;

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
10 rows affected.


[('China', Decimal('114085169487.10'), Decimal('936062666174.60'), Decimal('12.19'), 1),
 ('Brazil', Decimal('107043493413.30'), Decimal('936062666174.60'), Decimal('11.44'), 2),
 ('Russian Federation', Decimal('80519587149.10'), Decimal('936062666174.60'), Decimal('8.60'), 3),
 ('South Asia', Decimal('64687856286.30'), Decimal('936062666174.60'), Decimal('6.91'), 4),
 ('Turkey', Decimal('63650599648.00'), Decimal('936062666174.60'), Decimal('6.80'), 5),
 ('Mexico', Decimal('44486470550.40'), Decimal('936062666174.60'), Decimal('4.75'), 6),
 ('India', Decimal('43342982866.60'), Decimal('936062666174.60'), Decimal('4.63'), 7),
 ('Indonesia', Decimal('41106210447.60'), Decimal('936062666174.60'), Decimal('4.39'), 8),
 ('Kazakhstan', Decimal('32815594084.30'), Decimal('936062666174.60'), Decimal('3.51'), 9),
 ('Least developed countries: UN classification', Decimal('32282094104.60'), Decimal('936062666174.60'), Decimal('3.45'), 10)]

In [23]:
%%sql
SELECT country_name,
       ROUND(SUM(debt), 2) AS debt_inflow,
       ROUND(SUM(SUM(debt)) OVER(), 2) AS world_debt_inflow,
       ROUND((SUM(debt) / (SUM(SUM(debt)) OVER()) * 100), 2) AS pct_world_debt_inflow,
       RANK() OVER(ORDER BY SUM(debt) DESC) AS rank_inflow
  FROM (
        SELECT *
          FROM international_debt
         WHERE indicator_code = 'DT.DIS.DLXF.CD'
  ) AS subquery
 GROUP BY country_name
 ORDER BY debt_inflow DESC
 LIMIT 10;

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
10 rows affected.


[('Least developed countries: UN classification', Decimal('40160766261.60'), Decimal('264701069677.50'), Decimal('15.17'), 1),
 ('IDA only', Decimal('34531188113.20'), Decimal('264701069677.50'), Decimal('13.05'), 2),
 ('South Asia', Decimal('29306216064.70'), Decimal('264701069677.50'), Decimal('11.07'), 3),
 ('Cameroon', Decimal('18186662060.40'), Decimal('264701069677.50'), Decimal('6.87'), 4),
 ('China', Decimal('15692563746.10'), Decimal('264701069677.50'), Decimal('5.93'), 5),
 ('India', Decimal('11005547326.20'), Decimal('264701069677.50'), Decimal('4.16'), 6),
 ('Angola', Decimal('10924018093.10'), Decimal('264701069677.50'), Decimal('4.13'), 7),
 ('Egypt, Arab Rep.', Decimal('9552207423.50'), Decimal('264701069677.50'), Decimal('3.61'), 8),
 ('Bangladesh', Decimal('9050557611.90'), Decimal('264701069677.50'), Decimal('3.42'), 9),
 ('Vietnam', Decimal('6494121653.20'), Decimal('264701069677.50'), Decimal('2.45'), 10)]

6. Finding country debt outflow and inflow structure

In [17]:
%%sql
SELECT country_name, 
       indicator_code, 
       indicator_name,
       ROUND(SUM(debt), 2) AS debt_outflow,
       ROUND(SUM(SUM(debt)) OVER(), 2) AS world_debt_outflow,
       ROUND(SUM(debt) / SUM(SUM(debt)) OVER (PARTITION BY country_name) * 100, 2) AS pct_country_debt_outflow,
       ROUND((SUM(debt) / (SUM(SUM(debt)) OVER()) * 100), 2) AS pct_world_debt_outflow
  FROM (
        SELECT *
          FROM international_debt
         WHERE indicator_code IN ('DT.AMT.DPNG.CD', 'DT.AMT.OFFT.CD', 'DT.AMT.PRVT.CD', 'DT.INT.DPNG.CD', 'DT.INT.OFFT.CD', 'DT.INT.PRVT.CD')
  ) AS subquery
 GROUP BY country_name, indicator_code, indicator_name
 ORDER BY country_name, indicator_code
 LIMIT 10;

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
10 rows affected.


[('Afghanistan', 'DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('100847181.90'), Decimal('936062666174.60'), Decimal('65.45'), Decimal('0.01')),
 ('Afghanistan', 'DT.INT.OFFT.CD', 'PPG, official creditors (INT, current US$)', Decimal('53239440.10'), Decimal('936062666174.60'), Decimal('34.55'), Decimal('0.01')),
 ('Albania', 'DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('514185620.00'), Decimal('936062666174.60'), Decimal('53.79'), Decimal('0.05')),
 ('Albania', 'DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('236447897.30'), Decimal('936062666174.60'), Decimal('24.74'), Decimal('0.03')),
 ('Albania', 'DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('39615157.90'), Decimal('936062666174.60'), Decimal('4.14'), Decimal('0.00')),
 ('Albania', 'DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('87884000.00'), Decimal('936062666174.60'), Decimal('9.19'), Decimal('0.01')),
 ('Albania', 'DT.INT.OFFT.CD', 'PPG, official creditors (INT, current US$)', Decimal('41948869.70'), Decimal('936062666174.60'), Decimal('4.39'), Decimal('0.00')),
 ('Albania', 'DT.INT.PRVT.CD', 'PPG, private creditors (INT, current US$)', Decimal('35769517.20'), Decimal('936062666174.60'), Decimal('3.74'), Decimal('0.00')),
 ('Algeria', 'DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('75420000.00'), Decimal('936062666174.60'), Decimal('39.65'), Decimal('0.01')),
 ('Algeria', 'DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('95188724.60'), Decimal('936062666174.60'), Decimal('50.04'), Decimal('0.01'))]

In [16]:
%%sql
SELECT country_name, 
       indicator_code, 
       indicator_name,
       ROUND(SUM(debt), 2) AS debt_inflow,
       ROUND(SUM(SUM(debt)) OVER(), 2) AS world_debt_inflow,
       ROUND(SUM(debt) / SUM(SUM(debt)) OVER (PARTITION BY country_name) * 100, 2) AS pct_country_debt_inflow,
       ROUND((SUM(debt) / (SUM(SUM(debt)) OVER()) * 100), 2) AS pct_world_debt_inflow
  FROM (
        SELECT *
          FROM international_debt
         WHERE indicator_code IN ('DT.DIS.OFFT.CD', 'DT.DIS.PRVT.CD')
  ) AS subquery
 GROUP BY country_name, indicator_code, indicator_name
 ORDER BY country_name, indicator_code
 LIMIT 10;

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
10 rows affected.


[('Afghanistan', 'DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('72894453.70'), Decimal('255496114277.50'), Decimal('100.00'), Decimal('0.03')),
 ('Albania', 'DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('310371858.40'), Decimal('255496114277.50'), Decimal('97.85'), Decimal('0.12')),
 ('Albania', 'DT.DIS.PRVT.CD', 'PPG, private creditors (DIS, current US$)', Decimal('6822654.10'), Decimal('255496114277.50'), Decimal('2.15'), Decimal('0.00')),
 ('Algeria', 'DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('10320772.20'), Decimal('255496114277.50'), Decimal('100.00'), Decimal('0.00')),
 ('Angola', 'DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('9017246499.30'), Decimal('255496114277.50'), Decimal('82.55'), Decimal('3.53')),
 ('Angola', 'DT.DIS.PRVT.CD', 'PPG, private creditors (DIS, current US$)', Decimal('1906771593.80'), Decimal('255496114277.50'), Decimal('17.45'), Decimal('0.75')),
 ('Armenia', 'DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('426959175.60'), Decimal('255496114277.50'), Decimal('100.00'), Decimal('0.17')),
 ('Azerbaijan', 'DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('740907061.60'), Decimal('255496114277.50'), Decimal('68.07'), Decimal('0.29')),
 ('Azerbaijan', 'DT.DIS.PRVT.CD', 'PPG, private creditors (DIS, current US$)', Decimal('347550999.60'), Decimal('255496114277.50'), Decimal('31.93'), Decimal('0.14')),
 ('Bangladesh', 'DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('9050557611.90'), Decimal('255496114277.50'), Decimal('100.00'), Decimal('3.54'))]

7. The highest amount of principal repayments

In [49]:
%%sql
SELECT country_name,
       indicator_name,
       indicator_code,
       MAX(debt)
  FROM international_debt
 GROUP BY country_name, indicator_name, indicator_code
HAVING indicator_code = 'DT.AMT.DLXF.CD'
 ORDER BY MAX(debt) DESC;

 * postgresql://jupyter:***@kir4191-database.cv5qddmprgvd.us-east-2.rds.amazonaws.com:5432/intdebt
124 rows affected.


[('China', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('96218620835.699996948')),
 ('Brazil', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('90041840304.100006104')),
 ('Russian Federation', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('66589761833.5')),
 ('Turkey', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('51555031005.800003052')),
 ('South Asia', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('48756295898.199996948')),
 ('India', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('31923507000.799999237')),
 ('Indonesia', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('30916112653.799999237')),
 ('Kazakhstan', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('27482093686.400001526')),
 ('Mexico', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('25218503927')),
 ('Least developed countries: UN classification', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('25197029299.400001526')),
 ('IDA only', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('20483289208')),
 ('Romania', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('14013783350.399999619')),
 ('Colombia', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('11985674438.700000763')),
 ('Angola', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('11067045628.100000381')),
 ('Cameroon', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('10404814960.200000763')),
 ('Venezuela, RB', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('9878659207.200000763')),
 ('Egypt, Arab Rep.', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('9692114176.899999619')),
 ('Lebanon', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('9506919669.600000381')),
 ('South Africa', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('9474257551.899999619')),
 ('Vietnam', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('8873505909.200000763')),
 ('Pakistan', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('8336013891.300000191')),
 ('Ukraine', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('8148995625.600000381')),
 ('Philippines', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('6479170917.699999809')),
 ('Thailand', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('5914706998.300000191')),
 ('Peru', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('5881259974.699999809')),
 ('Sri Lanka', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('5761489255.899999619')),
 ('Morocco', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('5153430846.800000191')),
 ('Belarus', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('4640037884.199999809')),
 ('Ecuador', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('3623731570.1999998